source/extensions/quic_listeners/quiche/active_quic_listener.cc

```diff
@@ -112,14 +115,32 @@ void ActiveQuicListener::onData(Network::UdpRecvData& data) {
                                   /*packet_headers=*/nullptr, /*headers_length=*/0,
                                   /*owns_header_buffer*/ false);
   quic_dispatcher_->ProcessPacket(self_address, peer_address, packet);
+
+  // On event loop runs where this listener reads or has packets posted to it, process buffered
+  // packets on the next loop. This ensures that the limit of processed client hellos per loop is
+  // honored, and that buffered client hellos are processed in a timely fashion.
+  if (quic_dispatcher_->HasChlosBuffered()) {
+    udp_listener_->activateRead();
+  }
 }
 
 void ActiveQuicListener::onReadReady() {
+
   if (!enabled_.enabled()) {
```

Comment:
There is no need to pass in `enabled_` to EnvoyQuicDispather to disable QuicListener. You can make an early return like this in `ActiveQuicListener::onDataWorker()`

Reply:
I wasn't sure what the purpose of the runtime disable was. I was trying to maintain semantics of not allowing new sessions, but continuing to process packets on established sessions.

If that's overkill, I'm happy to simplify it as you suggest here.